### PR TITLE
Remove StorageClusterInitialization

### PR DESCRIPTION
**Pull request by jarrpa**
We want finer control over which resources we reconcile and which we don't. This PR removes the need for a StorageClusterInitialization CR by implementing a set of string to control how a given type of resource is reconciled. The default is set to only reconcile if it doesn't exist, to maintain compatibility with current behavior on upgrade.

### CODE REVIEW

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+// ReconcileStratery is a string representing how we want to reconcile
+// (or not) a particular resource
+type ReconcileStratery string
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/reconcile.go:30]: `s/ReconcileStatery/ReconcileStrategy/` everywhere?

**Review by obnoxxx — CHANGES_REQUESTED** [review]
General direction looks great.
First thing that caught my eye: `ReconcileStratery` ...

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (jarrpa) [reply pkg/controller/storagecluster/reconcile.go]: HAH! At least the typo was consistent ;D Fixed.

Comment (obnoxxx) [reply pkg/controller/storagecluster/reconcile.go]: yeah, it built...

**Review by obnoxxx — APPROVED** [review]
/lgtm

thanks for the update

**Pull request merged**